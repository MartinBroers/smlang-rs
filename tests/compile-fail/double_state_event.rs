extern crate smlang;

use smlang::statemachine;

statemachine! {
    transitions: {
        *State1 + Event1 = State2,
        State1 + Event1 = State3, //~ State and event combination specified multiple times, remove duplicates.
    }
}

fn main() {}
