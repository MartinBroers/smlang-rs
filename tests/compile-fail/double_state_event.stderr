error: State1 + Event1: State and event combination specified multiple times, remove duplicates.
  --> tests/compile-fail/double_state_event.rs:5:1
   |
5  | / statemachine! {
6  | |     transitions: {
7  | |         *State1 + Event1 = State2,
8  | |         State1 + Event1 = State3, //~ State and event combination specified multiple times, remove duplicates.
9  | |     }
10 | | }
   | |_^
   |
   = note: this error originates in the macro `statemachine` (in Nightly builds, run with -Z macro-backtrace for more info)
